modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -133,17 +135,17 @@ object KafkaConsumer {
         val chunkQueue: F[Queue[F, Option[Chunk[CommittableConsumerRecord[F, K, V]]]]] =
           Queue.bounded(settings.maxPrefetchBatches - 1)
 
-        type PartitionRequest =
+        type PartitionResult =
           (Chunk[KafkaByteConsumerRecord], FetchCompletedReason)
 
         type PartitionsMap = Map[TopicPartition, Stream[F, CommittableConsumerRecord[F, K, V]]]
         type PartitionsMapQueue = Queue[F, Option[PartitionsMap]]
 
-        def createPartitionStream(
+        def partitionStream(
           streamId: StreamId,
           partition: TopicPartition,
           assignmentRevoked: F[Unit]
-        ): F[Stream[F, CommittableConsumerRecord[F, K, V]]] =
+        ): Stream[F, CommittableConsumerRecord[F, K, V]] = Stream.force {
```

Comment:
This signature change means each partition stream will be initialized when the user starts pulling from it, rather than before the map is emitted - I don't see why that should be a problem, but if it is then this change can easily be reversed.